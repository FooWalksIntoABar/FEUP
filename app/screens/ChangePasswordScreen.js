import React from 'react'
import { View, Text, TextInput, TouchableOpacity } from 'react-native'

import styles from '../styles/ParentStyles/ChangePasswordScreen.style'

export default class ChangePasswordScreen extends React.Component {
  static navigationOptions = {
    headerMode: 'screen',
    navigationOptions: {
      headerTitle: 'Repor E-mail'
    }
  };
  render () {
    return (
      <View style={styles.generalLayout}>
        <View style={styles.container}>
          <Text>Insira a sua antiga password.</Text>
          <TextInput
            secureTextEntry
            placeholder='Insira o seu novo e-mail'
            style={styles.inputContainer}
          />
        </View>
        <View style={styles.container}>
          <Text>Insira a sua nova password.</Text>
          <TextInput
            secureTextEntry
            placeholder='Insira o seu novo e-mail'
            style={styles.inputContainer}
          />
        </View>
        <View style={styles.container}>
          <Text>Repita a sua nova password.</Text>
          <TextInput
            secureTextEntry
            placeholder='Insira o seu novo e-mail'
            style={styles.inputContainer}
          />
        </View>
        <View>
          <TouchableOpacity onPress={() => { console.log('changed password') }}>
            <Text>Alterar Password</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}
