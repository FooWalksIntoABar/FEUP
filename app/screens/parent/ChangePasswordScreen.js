import React from 'react'
import { View, Text, TextInput, TouchableOpacity } from 'react-native'

import styles from '../../styles/ParentStyles/ChangePasswordScreen.style'

export default class ChangePasswordScreen extends React.Component {
  static navigationOptions = {
    headerMode: 'screen',
    navigationOptions: {
      title: 'Alterar Password'
    }
  };
  render () {
    return (
      <View>
        <View style={styles.generalLayout}>
          <View style={{ paddingTop: 10, paddingBottom: 20 }}>
            <Text>Insira a sua antiga password.</Text>
            <TextInput
              secureTextEntry
              placeholder='Password Antiga'
              style={styles.inputContainer}
            />
          </View>
          <View style={styles.container}>
            <Text>Insira a sua nova password.</Text>
            <TextInput
              secureTextEntry
              placeholder='Nova Password'
              style={styles.inputContainer}
            />
          </View>
          <View style={styles.container}>
            <Text>Repita a sua nova password.</Text>
            <TextInput
              secureTextEntry
              placeholder='Nova Password'
              style={styles.inputContainer}
            />
          </View>
          <View style={styles.confirmBtn}>
            <TouchableOpacity onPress={() => { console.log('changed password') }}>
              <Text>Alterar Password</Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    )
  }
}
