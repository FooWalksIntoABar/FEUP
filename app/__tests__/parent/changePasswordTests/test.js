import React from 'react'
import { TouchableOpacity } from 'react-native'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import ChangePasswordScreen from '../../../screens/parent/ChangePasswordScreen'

configure({ adapter: new Adapter() })

describe('ChangePasswordScreen snapshot', () => {
  it('renders ChangePasswordScreen correctly', () => {
    const wrapper = shallow(<ChangePasswordScreen
      navigation={{ navigate: jest.fn() }} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find(TouchableOpacity).at(0).props().onPress()
    wrapper.unmount()
  })
})
