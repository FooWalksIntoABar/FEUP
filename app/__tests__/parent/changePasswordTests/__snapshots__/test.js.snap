// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePasswordScreen snapshot renders ChangePasswordScreen correctly 1`] = `
<Component>
  <Component
    style={
      Object {
        "padding": 60,
      }
    }
  >
    <Component
      style={
        Object {
          "paddingBottom": 20,
          "paddingTop": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Insira a sua antiga password.
      </Text>
      <TextInput
        allowFontScaling={true}
        placeholder="Password Antiga"
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "white",
            "borderWidth": 1,
            "height": 50,
            "marginTop": 20,
            "padding": 10,
          }
        }
      />
    </Component>
    <Component
      style={
        Object {
          "paddingBottom": 20,
          "paddingTop": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Insira a sua nova password.
      </Text>
      <TextInput
        allowFontScaling={true}
        placeholder="Nova Password"
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "white",
            "borderWidth": 1,
            "height": 50,
            "marginTop": 20,
            "padding": 10,
          }
        }
      />
    </Component>
    <Component
      style={
        Object {
          "paddingBottom": 20,
          "paddingTop": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Repita a sua nova password.
      </Text>
      <TextInput
        allowFontScaling={true}
        placeholder="Nova Password"
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "white",
            "borderWidth": 1,
            "height": 50,
            "marginTop": 20,
            "padding": 10,
          }
        }
      />
    </Component>
    <Component
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "orange",
          "borderRadius": 10,
          "height": 40,
          "justifyContent": "center",
          "shadowColor": "black",
          "shadowOpacity": 5,
          "shadowRadius": 10,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Alterar Password
        </Text>
      </TouchableOpacity>
    </Component>
  </Component>
</Component>
`;
